go/Features/logging/cmd/loggingServer/server.go

```diff
@@ -112,68 +113,99 @@ func resultsHandler(w http.ResponseWriter, r *http.Request) {
 }
 
 func processLogs() {
-    mu.Lock()
-    defer mu.Unlock()
-
-    if len(messageStore) == 0 {
-        log.Println("No messages to process.")
-        return
-    }
-
-    dbConn, err := db.ConnectToDB()
-    if err != nil {
-        log.Fatalf("Could not connect to the database: %v", err)
-    }
-    defer dbConn.Close()
-
-    result := make(map[string]string)
-    success := true
-
-    for _, msg := range messageStore {
-        err = logger.SendLogToDB(dbConn, msg.LogLevel, msg.CurrentService, msg.SourceService, msg.TypeOfRequest, msg.Content)
-        if err != nil {
-            log.Printf("Failed to insert log: %v", err)
-            success = false
-            break
-        }
+	mu.Lock()
+	defer mu.Unlock()
+
+	if len(messageStore) == 0 {
```

Comment:
Do you think we should continue to use messageStore? Another possible way is to create a new routine / thread to send each log to RabbitMQ and this should resolve delay caused by locking. Also, we can create a connection pool with RabbitMQ. Please add TODO to track this issue.

Reply:
Did you check my comment and Ruju's one? Also, when asking questions, you should show your idea including several options and pros / cons of each option.